testing/matrix-sdk-integration-testing/src/tests/sliding_sync/room.rs

```diff
@@ -538,3 +550,299 @@ async fn test_room_notification_count() -> Result<()> {
 
     Ok(())
 }
+
+static DROP_TODEVICE: StdMutex<bool> = StdMutex::new(true);
+fn drop_todevice(response: &mut Bytes) {
+    let Ok(mut json) = serde_json::from_slice::<serde_json::Value>(response) else {
+        return;
+    };
+    let Some(extensions) = json.get_mut("extensions").and_then(|e| e.as_object_mut()) else {
+        return;
+    };
+    let Some(to_device) = extensions.remove("to_device") else {
+        return;
+    };
+    if *DROP_TODEVICE.lock().unwrap() {
+        info!("Dropping to_device: {to_device}");
+        *response = serde_json::to_vec(&json).unwrap().into();
+        return;
+    }
+}
+
+struct CustomResponder {
+    client: reqwest::Client,
+    response_preprocessor: fn(&mut Bytes),
+}
```

Comment:
nit: please add newlines after each new struct, impl block, if with a long body, for with a long body, meaningful statement or expression, etc. in this file :-)